totalRP3/modules/dashboard/DashboardAboutTabView.lua

```diff
@@ -0,0 +1,340 @@
+end
+
+--- Formats the given format string against the given table of replacements.
+--
+--  The format string can contain standard format specifiers as well as WoW
+--  ones ("%s", "%1$s") as well as named replacements in the similar form
+--  of "%KEY_NAME$s".
+--
+--  If the key segment of the specifier can be converted to a number, it
+--  will be. The key can otherwise only contain characters that would form a
+--  valid Lua identifier.
+--
+--  Positional replacements (%s) are looked up in the array part of the table
+--  and, for each match, an offset incremented. That is to say, the behavior
+--  between this and string.format() is be identical.
+--
+--  @param formatString The template string to format.
+--  @param replacements Table of replacements to make.
+local function stringFormatNamed(format, replacements)
+	local replacer = StringFormatter();
+	local formatted = replacer:Format(format, replacements);
+	replacer:Dispose();
+	return formatted;
+end
+
+--*-*-*-*-*-*-*-*-*-*-*-*-*-*-*-*-*-*-*-*
+-- CreditRoles Enum
+--*-*-*-*-*-*-*-*-*-*-*-*-*-*-*-*-*-*-*-*
+
+--- Enumeration of valid roles for use with the Credit class.
+local CreditRoles = {
+	AUTHOR = "AUTHOR",
+	CONTRIBUTOR = "CONTRIBUTOR",
+	COMMUNITY_MANAGER = "COMMUNITY_MANAGER",
+	TESTER = "TESTER",
+	GUILD_MEMBER = "GUILD_MEMBER",
+};
+
+--*-*-*-*-*-*-*-*-*-*-*-*-*-*-*-*-*-*-*-*
+-- Credit Class
+--*-*-*-*-*-*-*-*-*-*-*-*-*-*-*-*-*-*-*-*
+
+--- This class represents a single person to be displayed in the credits.
+local Credit = Class("TRP3_Credit");
+
+--- Initialises the credit with the given person's name string and a list of
+--  roles to assign them.
+function Credit:initialize(person, ...)
+	self.person = person;
+	self.roles = { ... };
+end
+
+--- Returns the name of the person this credit belongs to.
+function Credit:GetName()
+	return self.person;
+end
+
+--- Returns the full text to display for this role, which consists of the
+--  persons name and their role(s).
+function Credit:GetLocalizedFullText()
+	local formatString = loc("THANK_YOU_WITH_ROLE");
+	return formatString:format(self:GetName(), self:GetLocalizedRoleText());
+end
+
+--- Returns a localized string that describes the roles this person has.
+function Credit:GetLocalizedRoleText()
+	-- Zero or only one role? Simple lookup, nothing more.
+	local roleCount = #self.roles;
+	if roleCount == 0 then
+		return "";
+	elseif roleCount == 1 then
+		return loc("THANK_YOU_ROLE_" .. self.roles[1]);
+	end
+
+	-- If this person is multi-talented and can perform not one, but MULTIPLE
+	-- roles then we need to join them up. Now, in all the supported locales
+	-- there is GUILD_MEMBER_TEMPLATE which exactly matches "%s, %s",
+	-- however we'll use our own GEN_LIST_DELIMITER just in case that one
+	-- ever changes purpose in some way.
+
+	-- Grab the first entry and then start the loop from the second index.
+	local output = loc("THANK_YOU_ROLE_" .. self.roles[1]);
+	for i = 2, roleCount do
```

Comment:
Couldn't this be simplified by adding all the localized roles in a table and doing a `table.concat(roles, ", ")`?

Reply:
I'd actually made a change here last minute and didn't think of the simple solution of `table.concat(roles, LIST_DELIMITER)`, good catch.

`LIST_DELIMITER` is defined in the global strings as `%s, %s` but seems to get changed somewhere along the line to being `, `. I'd prefer using that localised constant for the delimiter just in case tomorrow the Queen decrees that henceforth all list items should be separated by banana emoji.